bundles/ch.elexis.agenda/src/ch/elexis/agenda/preferences/AgendaDefinitionPreferencePage.java

```diff
@@ -274,6 +292,111 @@ protected void performApply() {
 		super.performApply();
 	}
 
+	private void emailConfirmationsGroup(Composite parentComposite) {
+		Group emailConfirmationsGroup = new Group(parentComposite, SWT.NONE);
+		emailConfirmationsGroup.setText(Messages.AgendaStandard_E_Mail_Title);
+		emailConfirmationsGroup.setLayout(new GridLayout(2, false));
+		emailConfirmationsGroup.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false, 2, 1));
+		new Label(emailConfirmationsGroup, SWT.NONE).setText(Messages.AgendaStandard_E_Mail);
+		accountsViewer = new ComboViewer(emailConfirmationsGroup);
+		accountsViewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
+		accountsViewer.setContentProvider(new ArrayContentProvider());
+		accountsViewer.setLabelProvider(new LabelProvider());
+		accountsViewer.addSelectionChangedListener(new ISelectionChangedListener() {
+			@Override
+			public void selectionChanged(SelectionChangedEvent event) {
+			}
+		});
+		Label emailTemplatesLabel = new Label(emailConfirmationsGroup, SWT.NONE);
+		emailTemplatesLabel.setText(Messages.AgendaStandard_E_Mail_Template);
+		emailTemplatesLabel.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 1, 1));
+		LabelProvider templateLabelProvider = new LabelProvider() {
+			@Override
+			public String getText(Object element) {
+				if (element instanceof ITextTemplate) {
+					return ((ITextTemplate) element).getName() + (((ITextTemplate) element).getMandator() != null
```

Comment:
Mach bitte einmal den cast (ITextTemplate) element) und weiß es zu, danach damit arbeiten, ist leichter lesbar

Reply:
Bitte auch als Lambda implementieren

Reply:
Ich habe festgestellt, dass LabelProvider mehrere Methoden enthält und deshalb konnte ich keine Lambda-Funktion verwenden. Aber ich habe den Code vereinfacht, indem ich die Logik in eine separate Methode ausgelagert habe, anstatt eine anonyme Klasse zu verwenden.